tests/test_butler.py

```diff
@@ -544,5 +570,99 @@ def tearDown(self):
             shutil.rmtree(self.root, ignore_errors=True)
 
 
+@unittest.skipIf(not boto3, "Warning: boto3 AWS SDK not found!")
+@mock_s3
+class S3DatastoreButlerTestCase(PosixDatastoreButlerTestCase):
+    """S3Datastore specialization of a butler; an S3 storage Datastore +
+    a local in-memory SqlRegistry.
+    """
+    configFile = os.path.join(TESTDIR, "config/basic/butler-s3store.yaml")
+    fullConfigKey = None
+    validationCanFail = True
+
+    bucketName = 'anybucketname'
+    """Name of the Bucket that will be used in the tests. The name is read from
+    the config file used with the tests during set-up.
+    """
+
+    root = 'butlerRoot/'
+    """Root repository directory expected to be used in case useTempRoot=False.
+    Otherwise the root is set to a 20 characters long randomly generated string
+    during set-up.
+    """
+
+    datastoreStr = [f"datastore={root}"]
+    """Contains all expected root locations in a format expected to be
+    returned by Butler stringification.
+    """
+
+    datastoreName = ["S3Datastore@s3://{bucketName}/{root}"]
+    """The expected format of the S3Datastore string."""
+
+    registryStr = f"registry='sqlite:///:memory:'"
+    """Expected format of the Registry string."""
+
+    def genRoot(self):
+        """Returns a random string of len 20 to serve as a root
+        name for the temporary bucket repo.
+
+        This is equivalent to tempfile.mkdtemp as this is what self.root
+        becomes when useTempRoot is True.
+        """
+        rndstr = ''.join(
+            random.choice(string.ascii_uppercase + string.digits) for _ in range(20)
+        )
+        return rndstr + '/'
+
+    def setUp(self):
+        config = Config(self.configFile)
```

Comment:
Shouldn't this be a `ButlerConfig`? (So that the doubled up `datastore.datastore` are removed)

Reply:
Not sure how to make this `ButlerConfig` since I need to call `makeRepo` first line of which is

    if isinstance(config, (ButlerConfig, ConfigSubset)):
        raise ValueError("makeRepo must be passed a regular Config without defaults applied.")

I guess I could try and recast it as `Config` when I make the call?

Reply:
Leave it for now. I'll have a think. We shouldn't be doing `datastore.datastore` because it's confusing.